api-lib/sendSocialMessage.test.ts

```diff
@@ -0,0 +1,102 @@
+import * as nodeFetch from 'node-fetch';
+
+import { createCircle } from '../api-test/helpers';
+
+import { adminClient } from './gql/adminClient';
+import { sendSocialMessage } from './sendSocialMessage';
+import { Awaited } from './ts4.5shim';
+
+const fetchOrig = jest.requireActual('node-fetch').default;
+
+const expectedResponse = {
+  ok: false,
+  status: 404,
+  statusText: 'Not Found',
+  headers: new Headers(),
+  text: jest.fn().mockResolvedValue(''),
+  json: () =>
+    Promise.resolve({
+      success: false,
+      error: 'Unexpected Id',
+    }),
+} as unknown as nodeFetch.Response;
+
+describe('sendSocialMessage', () => {
+  const consoleSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+
+  let circle: Awaited<ReturnType<typeof createCircle>>;
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  test('Test Failed Telegram message', async () => {
+    circle = await createCircle(adminClient, {
+      telegram_id: 'coordinapeTest123',
+    });
+
+    jest
+      .spyOn(nodeFetch, 'default')
+      .mockImplementationOnce(fetchOrig)
```

Comment:
hmm, i don't understand why you're doing this--you shouldn't have the test send a real request

Reply:
as was discussed before we didn't want to send to discord/telegram bad requests just for testing especially if the tests may run multiple times every time someone posts a change

Reply:
yes, i agree. but doesn't this line mean "set the mock implementation to the real implementation of fetch"? doesn't that cause a real call to the remote URL?

OHH, i remember now, this is because you actually want the first request to go through, because the first request is generated by the `createCircle` helper, right?

Reply:
> yes, i agree. but doesn't this line mean "set the mock implementation to the real implementation of fetch"? doesn't that cause a real call to the remote URL?
> 
> OHH, i remember now, this is because you actually want the first request to go through, because the first request is generated by the `createCircle` helper, right?

yes, if mocked fetch implementation for all createCircle fetch will be mocked too.